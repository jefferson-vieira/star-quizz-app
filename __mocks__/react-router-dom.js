import React from 'react';

const router = require('react-router-dom');

router.BrowserRouter = ({ children }) => <div>{children}</div>;

module.exports = router;
