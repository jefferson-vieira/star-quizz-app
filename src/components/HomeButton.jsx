import React from 'react';
import { Link } from 'react-router-dom';

const HomeButton = () => (
  <div className="row justify-content-center">
    <Link to="/home" className="btn btn-outline-warning btn-lg" title="Home">
      Home
    </Link>
  </div>
);

export default HomeButton;
