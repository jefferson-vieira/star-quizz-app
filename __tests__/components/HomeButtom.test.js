import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, mount } from 'enzyme';

import HomeButton from '../../src/components/HomeButton';

describe('Testing HomeButton Component...', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<HomeButton />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should redirect to /home', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/random']}>
        <HomeButton />
      </MemoryRouter>
    );

    expect(wrapper.find('Link').props().to).toBe('/home');
  });
});
