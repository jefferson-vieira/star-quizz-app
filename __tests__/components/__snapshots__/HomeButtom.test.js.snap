// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing HomeButton Component... should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        className="btn btn-outline-warning btn-lg"
        replace={false}
        title="Home"
        to="/home"
      >
        Home
      </Link>,
      "className": "row justify-content-center",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Home",
        "className": "btn btn-outline-warning btn-lg",
        "replace": false,
        "title": "Home",
        "to": "/home",
      },
      "ref": null,
      "rendered": "Home",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          className="btn btn-outline-warning btn-lg"
          replace={false}
          title="Home"
          to="/home"
        >
          Home
        </Link>,
        "className": "row justify-content-center",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Home",
          "className": "btn btn-outline-warning btn-lg",
          "replace": false,
          "title": "Home",
          "to": "/home",
        },
        "ref": null,
        "rendered": "Home",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
